commons/src/main/java/org/infinispan/commons/marshall/ProtoStreamTypeIds.java

```diff
@@ -0,0 +1,81 @@
+package org.infinispan.commons.marshall;
+
+/**
+ * TypeIds used by protostream in place of FQN message/enum names to reduce payload size.
+ * <p>
+ * ONCE SET VALUES IN THIS CLASS MUST NOT BE CHANGED AS IT WILL BREAK BACKWARDS COMPATIBILITY.
+ * <p>
+ * Values must in the range 0..65535, as this is marked for internal infinispan use by the protostream project.
+ * <p>
+ * If message/enum types are no longer required, the variable should be commented instead of deleted.
+ *
+ * @author Ryan Emerson
+ * @since 10.0
+ */
+public interface ProtoStreamTypeIds {
+
+   int MIN_ID = 0;
+
+   // Commons range 1 -> 1000
```

Comment:
We should comment what ranges produce what payload sizes (ie. 1, 2 or 3 bytes). Pretty much everything can go in the 2 byte range(128-16383), except for heavily used things (WrappedByteArray, PutKeyValueCommand etc) should go in 1 byte (0-127) and rarely used things should go into the 3 byte range memcached (16384+).